### PR TITLE
 Add data type metadata for fields and operators to LFRic tutorials (part of #817)

### PR DESCRIPTION
**Pull request by TeranIvy**
Another PR towards fully resolving #817. It updates field and operator metadata in the `tutorial/practicals/LFRic` directory with the data type descriptors. It also updates the relevant tutorial documentation.

This PR can be reviewed and merged independently of #1044 (updates test kernels except for 3 of them left to ensure support for both the old and the new style field and operator metadata) and #1045 (updates examples), but needs to be merged before #1052.

### CODE REVIEW

**Comment by TeranIvy** [comment]
Tested all modified tutorials with `make transform`. Tested `building_code` tutorials with `make compile`. Everything works fine. Ready for review.

File: tutorial/practicals/LFRic/building_code/2_built_ins/README.md [hunk 1]
```diff
@@ -370,7 +370,7 @@ The metadata is very similar to a user-defined LFRic kernel metadata
 with one major difference: built-ins are called from the [PSy-layer](
 ../background/LFRic_structure.md#psy-layer) loops over degrees of
 freedom (*DoFs*) of fields in the built-in, hence metadata identifier
-for this way of looping, `DOFS`. This means that the fields passed to
+for this way of looping, `DOF`. This means that the fields passed to
```
Comment (arporter) [inline tutorial/practicals/LFRic/building_code/2_built_ins/README.md:373]: "...) from within loops over the degrees of freedom (*DoFs*) of the field arguments. Hence the `operates_on` metadata for this type of kernel has the `DOF` value. All fields passed to..."

File: tutorial/practicals/LFRic/single_node/code/scaled_matrix_vector_kernel_mod.F90 [hunk 2]
```diff
@@ -45,8 +45,7 @@ module scaled_matrix_vector_kernel_mod
 
 use argument_mod,            only : arg_type,                 &
                                     GH_FIELD, GH_OPERATOR,    &
-                                    GH_READ, GH_INC,          &
-                                    ANY_SPACE_1, ANY_SPACE_2, &
+                                    GH_REAL, GH_READ, GH_INC, &
```
Comment (arporter) [inline tutorial/practicals/LFRic/single_node/code/scaled_matrix_vector_kernel_mod.F90:48]: Aren't ANY_SPACE_1/2 still needed?

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good. I've taken the opportunity to suggest improvements to some of the tutorial text that got pushed on in a bit of a hurry last time :-)
Only affects tutorial content. `make test` in `tutorial/practicals/LFRic` is still fine.

File: tutorial/practicals/LFRic/building_code/1_simple_kernels/LFRic_kernel_structure.md [hunk 3]
```diff
@@ -89,20 +91,20 @@ Here the `<base_name>` is `setval_field_w0`.
...
 arguments that this kernel operates on:
-* A field argument (`GH_FIELD`) with the degrees of freedom (DoFs) on the
- `W0` function space that is being updated in this kernel (`GH_INC` access
-  for updating fields on continuous function spaces);
-* A `real`-valued scalar (`GH_REAL`) whose value is read (`GH_READ`) and
-  used to update the field with.
+* A `real`-valued field argument (`GH_FIELD`, `GH_REAL`) with the degrees of
+  freedom (DoFs) on the `W0` function space that is being updated in this kernel
+  (`GH_INC` access for updating fields on continuous function spaces);
+* A `real`-valued scalar (`GH_SCALAR`, `GH_REAL`) whose value is read
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/1_simple_kernels/LFRic_kernel_structure.md:106]: Done.

File: tutorial/practicals/LFRic/building_code/1_simple_kernels/LFRic_kernel_structure.md [hunk 4]
```diff
@@ -89,20 +91,20 @@ Here the `<base_name>` is `setval_field_w0`.
 The PSyclone metadata is stored as the *private* data within the definition
 of an individual kernel type. They come in two main forms: derived types
 (e.g. `arg_type` in the above example) and `integer` arguments (single-valued
-such as `iterates_over` above or arrays). Here we briefly explain the contents
-of `arg_type` and `iterates_over` metadata. For more information please refer
+such as `operates_on` above or arrays). Here we briefly explain the contents
+of `arg_type` and `operates_on` metadata. For more information please refer
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/1_simple_kernels/LFRic_kernel_structure.md:95]: Done.

File: tutorial/practicals/LFRic/building_code/1_simple_kernels/LFRic_kernel_structure.md [hunk 5]
```diff
@@ -89,20 +91,20 @@ Here the `<base_name>` is `setval_field_w0`.
...
-* A field argument (`GH_FIELD`) with the degrees of freedom (DoFs) on the
- `W0` function space that is being updated in this kernel (`GH_INC` access
-  for updating fields on continuous function spaces);
-* A `real`-valued scalar (`GH_REAL`) whose value is read (`GH_READ`) and
-  used to update the field with.
+* A `real`-valued field argument (`GH_FIELD`, `GH_REAL`) with the degrees of
+  freedom (DoFs) on the `W0` function space that is being updated in this kernel
+  (`GH_INC` access for updating fields on continuous function spaces);
+* A `real`-valued scalar (`GH_SCALAR`, `GH_REAL`) whose value is read
+  (`GH_READ`) and used to update the field with.
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/1_simple_kernels/LFRic_kernel_structure.md:107]: Done.

File: tutorial/practicals/LFRic/building_code/1_simple_kernels/part1/README.md [hunk 6]
```diff
@@ -29,15 +29,15 @@ setval_field_w0_kernel_mod.f90) in an editor. The `arg_type` [metadata](
...
+         arg_type(GH_SCALAR, GH_REAL, GH_READ)     &
          /)
 ```
 
-describes two arguments: a field (`GH_FIELD`) with the degrees of freedom
-(DoFs) on the `W0` function space to be updated (`GH_INC`) and a
-`real`-valued scalar (`GH_REAL`) whose value is read
-(`GH_READ`) and used to update the field with.
+describes two arguments: a `real`-valued field (`GH_FIELD`, `GH_REAL`) with
+the degrees of freedom (DoFs) on the `W0` function space to be updated
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/1_simple_kernels/part1/README.md:38]: Thanks for the suggestion, done.

File: tutorial/practicals/LFRic/building_code/1_simple_kernels/part2/README.md [hunk 7]
```diff
@@ -29,16 +29,17 @@ setval_field_any_kernel_mod.f90) in an editor. The `arg_type` [metadata](
...
+         arg_type(GH_FIELD,  GH_REAL, GH_INC, ANY_SPACE_1), &
+         arg_type(GH_SCALAR, GH_REAL, GH_READ)              &
          /)
 ```
 
-describes two arguments: a field (`GH_FIELD`) with the degrees of freedom
-(DoFs) on a generic function space (`ANY_SPACE_1`) to be updated
-(`GH_INC`) and a `real`-valued scalar (`GH_REAL`) whose value is read
-(`GH_READ`) and used to update the field with.
+describes two arguments: a `real`-valued field (`GH_FIELD`, `GH_REAL`)
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/1_simple_kernels/part2/README.md:38]: Done as in the `part1/README.md `.

File: tutorial/practicals/LFRic/building_code/2_built_ins/README.md [hunk 8]
```diff
@@ -309,7 +309,7 @@ The generated code for the specific mathematical operation here
       END DO
 ```
 
-The above built-in code call also illustrates how `iterates_over = DOFS`
+The above built-in code call also illustrates how `operates_on = DOF`
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/2_built_ins/README.md:312]: Done.

File: tutorial/practicals/LFRic/building_code/2_built_ins/README.md (hunk 1 above)
Comment (TeranIvy) [reply tutorial/practicals/LFRic/building_code/2_built_ins/README.md:373]: Thanks for the suggestion, done.

File: tutorial/practicals/LFRic/single_node/code/scaled_matrix_vector_kernel_mod.F90 (hunk 2 above)
Comment (TeranIvy) [reply tutorial/practicals/LFRic/single_node/code/scaled_matrix_vector_kernel_mod.F90:48]: Oh yes, I must have accidentally deleted this line. I combed through other incorrect instances of this kernel and there was one in `examples/lfric/eg11` so I corrected it here.

**Comment by TeranIvy** [comment]
Addressed review comments and merged the latest master. Ready for another review from @arporter.

**Review by arporter — APPROVED** [review]
All requested changes have been made. Tests and examples still OK.

**Pull request merged**